tests/drop_counters/test_drop_counters.py

```diff
@@ -320,21 +578,464 @@ def test_src_ip_is_loopback_addr(ptfadapter, duthost, setup, tx_dut_ports, pkt_f
+    @summary: Verify that packet with absent destination IP address is dropped and L3 drop counter incremented
+    """
+    dut_iface, ptf_tx_port_id, dst_mac, src_mac = get_test_ports_info(ptfadapter, duthost, setup, tx_dut_ports)
+
+    log_pkt_params(dut_iface, dst_mac, src_mac, "", pkt_fields["ipv4_src"])
+
+    pkt = testutils.simple_tcp_packet(
+        eth_dst=dst_mac, # DUT port
+        eth_src=src_mac, # PTF port
+        ip_src=pkt_fields["ipv4_src"], # PTF source
+        ip_dst="", # VM source
+        tcp_sport=pkt_fields["tcp_sport"],
+        tcp_dport=pkt_fields["tcp_dport"])
+
+    base_verification("L3", pkt, ptfadapter, duthost, ptf_tx_port_id, tx_dut_ports[dut_iface])
+
+    # Verify packets were not egresed the DUT
+    exp_pkt = expected_packet_mask(pkt)
+    testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=setup["neighbor_sniff_ports"])
+
+
+@pytest.mark.parametrize("ip_addr", ["ipv4", "ipv6"])
+def test_src_ip_is_multicast_addr(ptfadapter, duthost, setup, tx_dut_ports, pkt_fields, ip_addr):
+    """
+    @summary: Verify that packet with multicast source IP adress is dropped and L3 drop counter incremented
+    """
+    dut_iface, ptf_tx_port_id, dst_mac, src_mac = get_test_ports_info(ptfadapter, duthost, setup, tx_dut_ports)
+    ip_src = None
+
+    if ip_addr == "ipv4":
+        ip_src = "224.0.0.5"
+        pkt = testutils.simple_tcp_packet(
+            eth_dst=dst_mac, # DUT port
+            eth_src=src_mac, # PTF port
+            ip_src=ip_src,
+            ip_dst=pkt_fields["ipv4_dst"], # VM source
+            tcp_sport=pkt_fields["tcp_sport"],
+            tcp_dport=pkt_fields["tcp_dport"])
+    elif ip_addr == "ipv6":
+        if not pkt_fields["ipv6_dst"]:
+            pytest.skip("BGP neighbour with IPv6 addr was not found")
+        ip_src = "FF02:AAAA:FEE5::1:3"
+        pkt = testutils.simple_tcpv6_packet(
+            eth_dst=dst_mac, # DUT port
+            eth_src=src_mac, # PTF port
+            ipv6_src=ip_src,
+            ipv6_dst=pkt_fields["ipv6_dst"], # VM source
+            tcp_sport=pkt_fields["tcp_sport"],
+            tcp_dport=pkt_fields["tcp_dport"])
+    else:
+        pytest.fail("Incorrect value specified for 'ip_addr' test parameter. Supported parameters: 'ipv4' and 'ipv6'")
+
+    log_pkt_params(dut_iface, dst_mac, src_mac, pkt_fields["ipv4_dst"], ip_src)
+    base_verification("L3", pkt, ptfadapter, duthost, ptf_tx_port_id, tx_dut_ports[dut_iface])
+
+    # Verify packets were not egresed the DUT
+    exp_pkt = expected_packet_mask(pkt)
+    testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=setup["neighbor_sniff_ports"])
+
+
+def test_src_ip_is_class_e(ptfadapter, duthost, setup, tx_dut_ports, pkt_fields):
+    """
+    @summary: Verify that packet with source IP address in class E is dropped and L3 drop counter incremented
+    """
+    dut_iface, ptf_tx_port_id, dst_mac, src_mac = get_test_ports_info(ptfadapter, duthost, setup, tx_dut_ports)
```

Comment:
This is also repeated everywhere - is there anyway we can make this a fixture or include it in pkt_fields?

Reply:
It is bad idea to move this into "pkt_fields", because pkt_fields calls only once per module and retrieves ansible facts which takes time. It will increase test execution time.

As a case it can be moved to some separate fixture.
What do you think?

Reply:
I see what you mean now! In that case, a separate fixture would be a good idea.

Reply:
Fixed.